src/modules/job-manager/jobtap.c

```diff
@@ -1563,36 +1589,113 @@ static struct job * jobtap_lookup_active_jobid (flux_plugin_t *p,
     return job;
 }
 
+/* Job aux items are not stored in the job aux container directly, to avoid
+ * segfaults that might result from registering a destructor resident in a
+ * plugin that could be unloaded before the item is destroyed.
+ *
+ * Instead, each plugin stores one item named "jobtap::<uuid>" which contains
+ * an aux container, and the actual items are stored in the inner container.
+ * As plugin is being unloaded, the outer container is destroyed before
+ * the plugin is destroyed, causing the inner container and its items to be
```

Comment:
The commit message still notes that jobtap aux items are leaked on unload, but this isn't the case anymore according to this comment.